spec/transactions/users/create_transaction_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+require 'dry/transaction'
+require_relative '../../../app/transactions/users/create_transaction'
+
+RSpec.describe "User::CreateTransaction", type: :transaction do
+  context "Create a user" do
+    it "should create a user if valid attributes given" do
+      expect { ::Users::CreateTransaction.call(valid_user_attributes) }
+        .to change { User.count }.by(1)
+    end
+
+    it "should not create a user if invalid attributes given" do
+      create(:user, email: 'nathan.patreau@capsens.fr')
+      expect { ::Users::CreateTransaction.call(invalid_user_attributes) }
+        .not_to change { User.count }
+    end
+  end
+
+  context "Send welcome mail", type: :mailer do
+    it "should send an email to user email" do
+      expect { ::Users::CreateTransaction.call(valid_user_attributes) }
+        .to change { ActionMailer::Base.deliveries.count }.by(1)
+      expect(ActionMailer::Base.deliveries.last.subject).to include('Welcome')
+    end
+
+    it "should raise an error if email not send" do
+      create(:user, email: 'nathan.patreau@capsens.fr')
+      expect { ::Users::CreateTransaction.call(invalid_user_attributes) }
+        .not_to change { ActionMailer::Base.deliveries.count }
+    end
+  end
+
+  def valid_user_attributes
+    {
+      first_name: 'Nathan',
+      last_name: 'Patreau',
+      email: 'nathan.patreau@capsens.fr',
+      birthdate: '08/09/1998',
+      password: '12345678'
+    }
+  end
+
+  def invalid_user_attributes
+    {
+      last_name: 'Patreau',
+      email: 'nathan.patreau@capsens.fr',
+      birthdate: '08/09/1998',
+      password: '12345678'
+    }
+  end
+end
```

Comment:
Utilise `subject` pour ne pas répéter à chaque fois "`expect { ::Users::CreateTransaction.call(invalid_user_attributes) }`"

Reply:
Je ne connais pas je vais regarder ça (j'utilisais Rails minitest)

Reply:
J'ai bien compris comment m'en servir et ça marche bien SAUF quand tu utilises une variable crée par un `let` dans ton `describe`. Le `let` n'autorise l'utilisation de sa variable crée que dans un `it`, un autre `let` ou un `before`.
Donc ce code retourne une erreur :
```
describe ::Users::CreateTransaction.call(valid_user_attributes) do
   it { is_expected.to change { ActionMailer::Base.deliveries.count }.by(1) }
 end
```
J'ai préféré ne pas l'utiliser, mais dis moi s'il y a un autre moyen !

Reply:
```ruby
describe ::Users::CreateTransaction do
  subject { described_class.call(user_attributes) }
```

et normalement ça va tout seul

Reply:
```ruby
it { expect(subject).to change { ActionMailer::Base.deliveries.count }.by(1) }
```

Reply:
Bon j'ai fait du mieux que j'ai pu, dis moi si tu vois une solution de faire plus condensé que ça.

Reply:
Bon j'ai fait du mieux que j'ai pu, dis moi si tu vois une solution de faire plus condensé que ça.

Reply:
Done ENFIN